libs/feature/teaching/src/lib/json-editor-dialog.tsx

```diff
@@ -12,12 +12,22 @@ export function JsonEditorDialog<T>({
 	onClose: (value: T | undefined) => void;
 }) {
 	const { getValues } = useFormContext();
-	const [value, setValue] = useState(JSON.stringify(getValues()));
+	const [jsonValue, setJsonValue] = useState(JSON.stringify(getValues()));
+	const [initialJsonValue] = useState(JSON.stringify(getValues()));
```

Comment:
es wäre effizienter und mehr straight forward, nicht den ganzen wert zu speichern, sondern nur die information ob der wert bereits geändert wurde (boolean).


```suggestion
	const [hasChanged, setHasChanged] = useState(false);
```

Reply:
Ich wüsste nicht wie, ohne den Editor stark zu verändern. Der Editor formatiert die JSON zu Beginn (Wird ohne \n etc eingefügt und dann erst im Editor richtig formatiert). Dabei wird die onChange-Methode aufgerufen, in der ich den Boolean setzen würde. Der Boolean wäre damit immer True. Mann müsste dann diesen Teil irgendwie vorziehen: ```TypeScript 
			onMount={editor => {
				editor.focus();
				setTimeout(() => {
					editor?.getAction("editor.action.formatDocument")?.run();
				}, 250);
			}}
```